tests/ecs_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <gtest/gtest.h>
+
+#include "ecs/game_coordinator.h"
+#include "ecs/components/transform.h"
+#include "ecs/systems/physics_system.h"
+
+namespace arch::ecs {
+	TEST(ECSTest, MovingObjects) {

```

Comment:
This test is maybe good enough for testing if a programmer uses ECS as he should, but it doesn't check what happens when he does something dumb. Maybe extract the data initialization to [test fixture](https://google.github.io/googletest/primer.html#same-data-multiple-tests) and then make a couple of more tests that check whether exceptions are thrown etc.

Reply:
Yeah I'm gonna add test in next steps, also after adding exceptions